addons/stock/models/stock_quant.py

```diff
@@ -341,6 +429,46 @@ def _quant_tasks(self):
         self._merge_quants()
         self._unlink_zero_quants()
 
+    def write(self, vals):
+        if 'inventory_quantity' in vals and not self.env.context.get('dont_create_inventory_line'):
+            for quant in self:
+                rounding = quant.product_id.uom_id.rounding
+                diff = float_round(vals['inventory_quantity'] - quant.quantity, precision_rounding=rounding)
+                diff_float_compared = float_compare(diff, 0, precision_rounding=rounding)
+                if diff_float_compared == 0:
+                    continue
+                elif diff_float_compared > 0:
+                    move = self.env['stock.move'].create(self._get_inventory_move_values(self.product_id.property_stock_inventory, self.location_id, diff))
+                    move._action_done()
+                else:
+                    move = self.env['stock.move'].create(self._get_inventory_move_values(self.location_id, self.product_id.property_stock_inventory, -1 * diff))
+                    move._action_done()
+            vals.pop('inventory_quantity')
+        return super(StockQuant, self).write(vals)
+
+    @api.model
+    def create(self, vals):
+        if 'inventory_quantity' in vals and not self.env.context.get('dont_create_inventory_line'):
+            if 'product_id' in vals:
+                product = self.env['product.product'].browse(vals['product_id'])
+            else:
+                product = self._default_product_id()
+            location = self.env['stock.location'].browse(vals['location_id'])
+            lot_id = self.env['stock.production.lot'].browse(vals.get('lot_id'))
+            package_id = self.env['stock.quant.package'].browse(vals.get('package_id'))
+            owner_id = self.env['res.partner'].browse(vals.get('owner_id'))
+            similar = self._gather(product, location, lot_id=lot_id, package_id=package_id, owner_id=owner_id, strict=True)
+            if similar:
+                similar.write({'inventory_quantity': vals['inventory_quantity']})
+                return similar
```

Comment:
i think it's really really worth having a comment here explaining that create sometimes does a write